pam/pam_test/module-transaction-dummy_test.go

```diff
@@ -0,0 +1,251 @@
+package pam_test
+
+import (
+	"testing"
+
+	"github.com/msteinert/pam"
+	"github.com/stretchr/testify/require"
+)
+
+type binConvFunc func(pam.BinaryPointer) ([]byte, error)
+
+func (f binConvFunc) RespondPAMBinary(ptr pam.BinaryPointer) ([]byte, error) {
+	return f(ptr)
+}
+
+func (f binConvFunc) RespondPAM(pam.Style, string) (string, error) {
+	return "", pam.ConvErr
+}
+
+func TestModuleTransactionDummy(t *testing.T) {
+	t.Parallel()
+	t.Cleanup(MaybeDoLeakCheck)
+
+	testCases := map[string]struct {
+		testFunc    func(*testing.T, pam.ModuleTransaction)
+		convHandler pam.ConversationHandler
+	}{
+		"SetItem": {
+			testFunc: func(t *testing.T, tx pam.ModuleTransaction) {
+				t.Helper()
+				err := tx.SetItem(pam.User, "an user")
+				require.NoError(t, err, "SetItem should not return an error")
+			},
+		},
+
+		"GetItem Empty": {
+			testFunc: func(t *testing.T, tx pam.ModuleTransaction) {
+				t.Helper()
+				val, err := tx.GetItem(pam.User)
+				require.NoError(t, err, "GetItem should not return an error")
+				require.Equal(t, "", val)
+			},
+		},
+
+		"GetSetItem": {
+			testFunc: func(t *testing.T, tx pam.ModuleTransaction) {
+				t.Helper()
+				err := tx.SetItem(pam.User, "an user")
+				require.NoError(t, err, "SetItem should not return an error")
+				val, err := tx.GetItem(pam.User)
+				require.NoError(t, err, "GetItem should not return an error")
+				require.Equal(t, "an user", val)
+			},
+		},
```

Comment:
> especially the PutEnvGetEnv one.  It has multiple assertions to test different use cases and those could easily become test cases inside a table test and then the function itself would need only to assert a single time on the values returned by PutEnv and GetEnv (easier to debug as well when a test fails, since the logs will show the complete name of the failed test (e.g. TestSomething/SubTestOfTestSomething).

One reason I've not used that was because such tests are not parellizable, so having-sub tests wasn't a big advantage from my POV.

I'll address those though.